modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -1372,10 +1373,11 @@ public void addServiceBody(Set<Whitespace> ws) {
                 .forEach(varDef -> serviceNode.addVariable((VariableDefinitionNode) varDef));
     }
 
-    public void endServiceDef(Set<Whitespace> ws, String protocolPkg, String serviceName) {
+    public void endServiceDef(DiagnosticPos pos, Set<Whitespace> ws, String protocolPkg, String serviceName) {
         ServiceNode serviceNode = serviceNodeStack.pop();
         serviceNode.setName(createIdentifier(serviceName));
         serviceNode.setProtocolPackageIdentifier(createIdentifier(protocolPkg));
+        ((BLangService) serviceNode).pos = pos;
```

Comment:
Change the "serviceNode" variable to BLangService itself when it's created at 1377, by doing the explicit cast there.

Reply:
Fixed